src/formats/formats.js

```diff
@@ -37,40 +37,42 @@ export function formatByExtension(extension) {
   }[extension];
 }
 
-function formatByName(name) {
+function formatByName(name, customDelimiter) {
   return {
     yml: yamlFormatter,
     yaml: yamlFormatter,
     toml: tomlFormatter,
     json: jsonFormatter,
     frontmatter: FrontmatterInfer,
-    'json-frontmatter': FrontmatterJSON,
-    'toml-frontmatter': FrontmatterTOML,
-    'yaml-frontmatter': FrontmatterYAML,
+    'json-frontmatter': frontmatterJSON(customDelimiter),
+    'toml-frontmatter': frontmatterTOML(customDelimiter),
+    'yaml-frontmatter': frontmatterYAML(customDelimiter),
   }[name];
 }
 
 export function resolveFormat(collectionOrEntity, entry) {
+  // Check for custom delimiter
+  const customDelimiter = collectionOrEntity.get('frontmatter_delimiter');
```

Comment:
I think we need to throw if a `frontmatter_delimiter` is set without an explicit `format` being set. This might be better checked in the reducer config validation section: https://github.com/netlify/netlify-cms/blob/e9ec6def04d8a10bcf28173780d9939b30bc195e/src/reducers/collections.js#L29-L41

Reply:
@Swieckowski I think we discussed this before, was there something that I missed?

Reply:
Ah yes, I remember bringing that up myself but don't recall a specific response, sorry. Should I get on that asap?

Reply:
No problem! I'd like to have it in before merge -- if you're busy, I'd be glad to do it.

Reply:
I can have it done by 2 or 3pm tomorrow if that's alright.

Reply:
Oh, no hurry at all, thank you!

Reply:
@tech4him1 I've been a bit swamped with job search stuff haha, but I hope my latest commits address what you were talking about. We now throw an error if someone has a frontmatter_delimiter set without setting their format or not setting it to the proper frontmatter formats.

Reply:
Also, since I've added tests and documentation, how do I add that to my contribution? The add contributor script is failing on me, perhaps because I've already run it once on my github username?